assets/src/settings-page/supported-templates-visibility.js

```diff
@@ -0,0 +1,69 @@
+/**
+ * WordPress dependencies
+ */
+import { useRef, useEffect, useContext } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import { Options } from '../components/options-context-provider';
+
+/**
+ * Side effect component hiding and showing theme support form input fields depending on the state of other selections on the page.
+ *
+ * @todo The external DOM components this manipulates should eventually be converted to React.
+ */
+export function SupportedTemplatesVisibility() {
+	const { editedOptions } = useContext( Options );
+
+	const { all_templates_supported: allTemplatesSupported, theme_support: themeSupport } = editedOptions || {};
+
+	const supportedPostTypesTitle = useRef( document.querySelector( '#all_templates_supported_fieldset, #supported_post_types_fieldset > .title' ) );
+	const supportedPostTypesFieldset = useRef( document.getElementById( 'supported_post_types_fieldset' ) );
+	const supportedTemplatesFieldset = useRef( document.getElementById( 'supported_templates_fieldset' ) );
+	const supportedTemplateInputs = useRef( [ ...document.querySelectorAll( '#supported_templates_fieldset input[type=checkbox]' ) ] );
+
+	/**
+	 * Show/hide settings features depending on options on the page.
+	 */
+	useEffect( () => {
+		supportedPostTypesTitle.current.classList.toggle(
+			'hidden',
+			'reader' === themeSupport,
+		);
+
+		supportedPostTypesFieldset.current.classList.toggle(
+			'hidden',
+			allTemplatesSupported && 'reader' !== themeSupport,
+		);
+
+		supportedTemplatesFieldset.current.classList.toggle(
+			'hidden',
+			allTemplatesSupported || 'reader' === themeSupport,
```

Comment:
This needs to change so that it is only hidden if:

```diff
- allTemplatesSupported || 'reader' === themeSupport
+ allTemplatesSupported || ( 'reader' === themeSupport && 'legacy' === readerTheme )
```

In other words, this section should be hidden if all templates are supported _or_ the legacy Reader mode is active.

Reply:
I _think_ I got this worked out: e464cc2232ba6f025cbcfcc3602030123e1a2bad